tests/integration/integration_suite_test.go

```diff
@@ -1,13 +1,316 @@
+		cleanupCluster(t, namespace)
+	}
+}
+
+func deployFullStackCluster(t *testing.T, namespace string) {
+	DeployClusterWithValues(t, namespace, "minio", "three_nodes_cluster_full_stack.yaml")
+	checkResourcePresenceForReaper(t, namespace)
+	waitForReaperPod(t, namespace)
+	checkReaperRegistered(t, namespace)
+}
+
+// Reaper scenario:
+// - Install Traefik
+// - Create a namespace
+// - Register a cluster with 3 Cassandra nodes
+// - Verify that Reaper correctly initializes
+// - Start a repair on the reaper_db keyspace
+// - Wait for at least one segment to be processed
+// - Cancel the repair
+// - Terminate the namespace and delete the cluster
+func TestReaperDeploymentScenario(t *testing.T) {
+	// only runs when the -short flag is used
+	if !testing.Short() {
+		t.Skip("Skipped in non-short mode")
+	}
+	namespace := initializeCluster(t)
+	deployClusterForReaper(t, namespace)
+	testReaper(t, namespace)
+	cleanupCluster(t, namespace)
+}
+
+func testReaper(t *testing.T, namespace string) {
+	log.Println(Step("Testing Reaper..."))
+	repairId := triggerRepair(t, namespace)
+	waitForSegmentDoneAndCancel(t, repairId)
+}
+
+func deployClusterForReaper(t *testing.T, namespace string) {
+	log.Println(Info("Deploying K8ssandra and waiting for Reaper to be ready"))
+	DeployClusterWithValues(t, namespace, "default", "three_nodes_cluster_with_reaper.yaml")
+	checkResourcePresenceForReaper(t, namespace)
+	waitForReaperPod(t, namespace)
+	checkReaperRegistered(t, namespace)
+}
+
+func checkResourcePresenceForReaper(t *testing.T, namespace string) {
+	CheckResourceWithLabelIsPresent(t, namespace, "service", "app.kubernetes.io/managed-by=reaper-operator")
+	CheckClusterExpectedResources(t, namespace)
+}
+
+func waitForReaperPod(t *testing.T, namespace string) {
+	WaitForReaperPod(t, namespace)
+}
+
+func checkReaperRegistered(t *testing.T, namespace string) {
+	CheckKeyspaceExists(t, namespace, "reaper_db")
+	CheckClusterIsRegisteredInReaper(t, "k8ssandra")
+}
+
+func triggerRepair(t *testing.T, namespace string) string {
+	log.Println(Info("Starting a repair"))
+	return TriggerRepair(t, namespace, "reaper_db")
+}
+
+func waitForSegmentDoneAndCancel(t *testing.T, repairId string) {
+	log.Println(Info("Waiting for one segment to be repaired and canceling run"))
+	WaitForOneSegmentToBeDone(t, repairId)
+	CancelRepair(t, repairId)
+}
+
+// Medusa scenario (invoked with a specific backend name):
+// - Register a cluster with 1 node
+// - Potentially install backend specific dependencies (such as Minio)
+// - Create the backend credentials secret
+// - Create a keyspace and a table
+// - Load 10 rows and check that we can read 10 rows
+// - Perform a backup with Medusa
+// - Load 10 rows and check that we can read 20 rows
+// - Restore the backup
+// - Verify that we can read 10 rows
+// - Terminate the namespace and delete the cluster
+func TestMedusaDeploymentScenario(t *testing.T) {
+	// only runs when the -short flag is used
+	if !testing.Short() {
```

Comment:
Is this backwards? If tests are run with the `-short` flag, that indicates that the user wants to skip long running tests.

Reply:
The long test here is the full stack one, which will run only if the `-short` flag isn't used. If individual components tests scenarios are to run, then you'll need to specify the `-short` flag. 
The full stack test already checks all that the individual tests do, so it's redundant to run them all when invoking `go test` without specifying a test function.
This way, if you want to test everything (without running redundant tests), you can just run `go test -test.timeout=30m ./tests/integration -v` and if you want to run an individual scenario : `go test $(GO_FLAGS) -test.timeout=30m ./tests/integration -v -short -run=TestReaperDeploymentScenario`.

Do you know of any other trick that would allow us to just run `TestFullStackScenario` by default, while still allowing to run the other tests if needed?

Reply:
Thanks for the explanation. Rather than have every individual component test check `testing.Short()`, I would switch the behavior and have `TestFullStackScenario` do the check. This would mean all tests run by default which I think is good and I can opt out of longer running tests with`-short`.

Reply:
I've removed the use of `-short`. The Makefile handles running the appropriate tests only. Running `go test ...` would run all tests though, the full stack one and the individual ones as well.